app/controllers/local_authorities_controller.rb

```diff
@@ -21,6 +22,26 @@ def update
 
 private
 
+  def links_for_authority
+    @_links_for_authority ||= filtered_links
+      .includes([:service, :interaction])
+      .all
+  end
+
+  def filtered_links
+    # Returns the list of links for enabled services only.
+    links = @authority.links.joins(:service).merge(@authority.provided_services)
```

Comment:
There's a subtle difference between the two, which the code comment doesn't address. Now you've pointed it out, I'm not sure which is the right solution.

For example:
```
> @authority.links.enabled_links.count
SELECT COUNT(*) 
FROM "links" 
INNER JOIN "service_interactions" ON "service_interactions"."id" = "links"."service_interaction_id" 
INNER JOIN "services" ON "services"."id" = "service_interactions"."service_id" 
WHERE "links"."local_authority_id" = $1 
AND "services"."enabled" = $2  
[["local_authority_id", 41], ["enabled", true]]
=> 111
```

vs 

```
> @authority.links.joins(:service).merge(@authority.provided_services).count
SELECT COUNT(*) 
FROM "links" 
INNER JOIN "service_interactions" ON "service_interactions"."id" = "links"."service_interaction_id" 
INNER JOIN "services" ON "services"."id" = "service_interactions"."service_id" 
INNER JOIN "service_tiers" ON "services"."id" = "service_tiers"."service_id" 
WHERE "links"."local_authority_id" = $1 
AND "service_tiers"."tier_id" = $2 
AND "services"."enabled" = $3  
[["local_authority_id", 41], ["tier_id", 2], ["enabled", true]]
=> 97
```

In this case, the latter example produces fewer results.  Because it is only showing links for services of the appropriate Tier for the local authority.  It's interesting, as it shows we must have some out-of-date info in our database already.